Thunderbird 115 compatibility

jikamens opened the pull request:
This branch is based on the await-all-async branch (#563). It removes the custom Send Later column in the thread pane since it's not yet possible in TB15 for add-ons to add a custom column, and it restores the functionality of the Send Later header row in the message display, in a way that reduces the amount of code in the experiment in preparation for there eventually being a supported API for adding header rows.

There are a number of TODO's in this commit which I could use feedback on.

jikamens commented:
@jobisoft @arndissler I could definitely use your input here!

arndissler reviewed (changes requested):
We're getting close!

While the v115 (and v102) way of displaying additional information without an experiment is to use `messageDisplayScript`, I made some suggestions on the code to add a new header row in the message display.

When you're removing the APIs for the custom column, you can remove the files completely from the codebase, as they are no longer needed:
- `experiments/customColumnImplementation.js`
- `experiments/customColumnSchema.json`
- `experiments/legacyColumnImplementation.js`
- `experiments/legacyColumnSchema.json`

If we need the functionality, we can check the git history, but I guess if we wire up a new custom column, we need to implement it from scratch.

jikamens commented on experiments/hdrViewImplementation.js:
```diff
@@ -53,80 +50,63 @@ class CustomHdrRow {
+      // do document.createElement("simple-header-row") and have all this happen
+      // automatically!
+      // TODO The row heading and value are displaying on separate lines. I
+      // don't know why or what the right way to fix it is.
```
I tried your proposed HTML structure and it didn't work, either inside or outside `extraHeadersArea`.
However, I played around with it some more and managed to find a structure that works, by aping the structure and place in the hierarchy of the Organization header. The structure I settled on doesn't work inside `extraHeadersArea`, but it works nicely outside `extraHeadersArea`. See the commit I just pushed.

jikamens commented on experiments/hdrViewImplementation.js:
```diff
@@ -53,80 +50,63 @@ class CustomHdrRow {
+      heading.classList.add("row-heading");
+      let sep = document.createElement("span");
+      sep.classList.add("screen-reader-only");
+      sep.setAttribute("data-l10n-name", "field-separator");
```
Your second comment above about l10nName is moot because of the changes described below. Regarding the first comment I've pushed a commit deleting the obsolete files.

jikamens commented:
@arndissler thanks for your feedback, please have another look!

arndissler reviewed (approved):
Looks good, ready to merge. ☀️

Personally I suggest to add an auto formatter to the project, so the code style will be more consistent (like line breaks in `if-else` control structures before the `else`  and a missing block statement for single command blocks. Good pick for this task is https://prettier.io (isn't it the de-factor standard for JS nowadays?).

jikamens commented:
> Looks good, ready to merge. sunny
> 
> Personally I suggest to add an auto formatter to the project, so the code style will be more consistent (like line breaks in `if-else` control structures before the `else` and a missing block statement for single command blocks. Good pick for this task is https://prettier.io (isn't it the de-factor standard for JS nowadays?).

OK, I've now prettified everything, added config files to do prettier automatically in Emacs (the editor I use), and added a prettier Github workflow to enforce it on pull requests and pushes to main.

Oh, I also renamed master to main.
